pcdsdevices/state.py

```diff
@@ -692,6 +717,36 @@ class TwinCATStatePositioner(StatePositioner):
         doc='Configuration of state positions, deltas, etc.',
     )
 
+    def _get_state_velo(self, items: SignalToValue) -> float:
```

Comment:
This is a good point. Should these be defined outside of the class scope? I wonder what the best practices are...

Reply:
I guess the consequence of calling these outside of the MDS is pretty minimal, since they don't do any active reads/writes, so it's just up to `AttributeError` confusion mostly.

Reply:
Double underscore is pretty much made for that scenario. I'm not fond of its usage at all, personally.

As to best practices, I think it's a bit hard to see from where we're at. Maybe a couple more features using it and we'll have a better idea then?

Reply:
Maybe not right now, but the above is at least food for thought. Currently we ascribe special meaning to `self` (I'm not sure if python does?)
https://github.com/pcdshub/pcdsdevices/blob/5b2bd815cb16ec6fb23d2764b0be7e421a5a33eb/pcdsdevices/utils.py#L560

Reply:
I think you may be onto something there, @ZLLentz. The acceptable signature should change to avoid confusion and make this easier. 

One possibility would be a relaxed check of the function, similar to how it's done now:
1. Define it as taking two arguments, and you'll get the signal instance and items (/value)
2. Define it as taking a single argument, and you'll get just the items (/value)

Or we remove the flexibility and say "you must define it as taking two arguments", full-stop.  The flexible arrangement is nice for brevity / lambda functions, but in retrospect it's probably unnecessary complexity.

Reply:
Er, we should take this API discussion into an issue to save future confusion... Hold on...